paymentrequest/free-shipping/demo.js

```diff
@@ -33,8 +33,15 @@ function onBuyClicked() {
 
   var options = {requestShipping: true};
 
-  new PaymentRequest(supportedInstruments, details, options) // eslint-disable-line no-undef
-      .show()
+  var request = new PaymentRequest(supportedInstruments, details, options); // eslint-disable-line no-undef
+
+  request.addEventListener('shippingaddresschange', function(evt) {
+    evt.updateWith(new Promise(function(resolve) {
```

Comment:
Can you just use `Promise.resolve(details)` here and elsewhere instead of constructing a new `Promise`?

See http://stackoverflow.com/questions/23803743/what-is-the-explicit-promise-construction-antipattern-and-how-do-i-avoid-it